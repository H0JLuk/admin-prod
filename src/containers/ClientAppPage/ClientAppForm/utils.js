import { notification } from 'antd';
import { addSettings, updateSettingsList } from '../../../api/services/settingsService';
import { designKeysForCheck, SETTINGS_TYPES } from './ClientAppFormConstants';

export const showNotify = (message, isError) => {
    const config = {
        duration: 5,
        placement: 'bottomRight',
    };

    if (isError) {
        notification.error({
            ...config,
            message,
        });
    } else {
        notification.success({
            ...config,
            message,
        });
    }

};

export async function createOrUpdateKey (changedParams) {
    const updateSettings = [];
    const addSettingsArr = [];

    changedParams.forEach(({ type, ...params }) => {
        type === SETTINGS_TYPES.EDIT
            ? updateSettings.push(params)
            : addSettingsArr.push(params);
    });

    updateSettings.length && (await updateSettingsList(updateSettings));
    addSettingsArr.length && (await addSettings(addSettingsArr));
}

export const checkExistDesignSettings = (settings) => {
    return designKeysForCheck.some(key => !Object.prototype.hasOwnProperty.call(settings, key));
};

export const getImage = (imageName) => {
    return `${process.env.PUBLIC_URL}/images/clientAppDesign/${imageName}.png`;
};


