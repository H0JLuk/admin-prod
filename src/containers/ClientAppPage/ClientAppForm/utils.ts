import { customNotifications } from '@utils/notifications';
import { ArgsProps } from 'antd/lib/notification';
import { addSettings, updateSettingsList } from '@apiServices/settingsService';
import { SETTINGS_TYPES } from './ClientAppFormConstants';
import { SettingDto } from '@types';
import ROLES from '@constants/roles';

export interface IChangedParam extends Pick<SettingDto, 'clientAppCode' | 'key' | 'value' | 'userRole'> {
    type: SETTINGS_TYPES;
}

export const showNotify = (message: React.ReactNode, isError?: boolean) => {
    const config: ArgsProps = {
        duration: 5,
        message,
    };

    if (isError) {
        customNotifications.error(config);
    } else {
        customNotifications.success(config);
    }

};

type UpdateAndAddSettings = {
    updateSettings: SettingDto[];
    addSettingsArr: SettingDto[];
};

export async function createOrUpdateKey(changedParams: IChangedParam[]) {
    const { updateSettings, addSettingsArr } = changedParams.reduce<UpdateAndAddSettings>((result, { type, ...params }) => {
        const key = type === SETTINGS_TYPES.EDIT ? 'updateSettings' : 'addSettingsArr';

        if (params.key === 'referralTokenLifetime') {
            delete params.clientAppCode;
            params.key = 'token_lifetime';
            params.userRole = ROLES.REFERAL_LINK;
        }

        return {
            ...result,
            [key]: [...result[key], params],
        };
    }, { updateSettings: [], addSettingsArr: [] });

    updateSettings.length && (await updateSettingsList(updateSettings));
    addSettingsArr.length && (await addSettings(addSettingsArr));
}
