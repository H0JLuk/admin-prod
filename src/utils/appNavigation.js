import { ROUTE } from '../constants/route';
import ROLES from '../constants/roles';

export const goToLogin = (history) => {
    if (history) {
        history.push(ROUTE.LOGIN);
        return;
    }
    window.location.href = ROUTE.LOGIN;
};

export const goToClientApps = (history) => {
    history.push(ROUTE.CLIENT_APPS);
};

export const goToAudit = (history) => {
    history.push(ROUTE.AUDITOR);
};

export const goToAdmin = (history) => {
    history.push(ROUTE.ADMIN);
};

export const goToProduct= (history) => {
    history.push(ROUTE.PRODUCT);
};

export const goApp = (history, role) => {
    switch (role) {
        case ROLES.AUDITOR:
            goToAudit(history);
            break;
        case ROLES.ADMIN:
            goToAdmin(history);
            break;
        case ROLES.PRODUCT_OWNER:
        default:
            goToProduct(history);
    }
};

export const goBack = (history) => {
    history.goBack();
};
