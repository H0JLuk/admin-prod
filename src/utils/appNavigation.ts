import { History } from 'history';
import { getRole } from '@apiServices/sessionService';
import {
    CONSENTS_PAGES,
    PROMO_CAMPAIGN_PAGES,
    ROUTE,
    ROUTE_ADMIN,
    ROUTE_OWNER,
    ROUTE_USER_MANAGER,
    ROUTE_PARTNER,
} from '@constants/route';
import ROLES from '@constants/roles';

export const goToLogin = (history: History) => {
    if (history) {
        history.push(ROUTE.LOGIN);
        return;
    }
    window.location.href = ROUTE.LOGIN;
};

export const goToDashboard = (history: History, role: ROLES) => {
    switch (role) {
        case ROLES.ADMIN:
            history.push(ROUTE_ADMIN.DASHBOARD);
            break;
        case ROLES.PRODUCT_OWNER:
            history.push(ROUTE_OWNER.DASHBOARD);
            break;
        default:
            break;
    }
};

export function goToStartPage(history: History, redirect: boolean, role: ROLES | null | '' = getRole()) {
    switch (role) {
        case ROLES.ADMIN: {
            const url = ROUTE_ADMIN.DASHBOARD;
            (redirect ? history.replace : history.push)(url);
            return;
        }
        case ROLES.PRODUCT_OWNER: {
            const url = ROUTE_OWNER.DASHBOARD;
            (redirect ? history.replace : history.push)(url);
            return;
        }
        case ROLES.AUDITOR: {
            goToAudit(history, redirect);
            return;
        }
        case ROLES.USER_MANAGER: {
            goToUserManager(history, redirect);
            return;
        }
        case ROLES.PARTNER: {
            goToPartner(history, redirect);
            return;
        }
        case null:
        case '': {
            goToLogin(history);
            return;
        }
        default:
            return;
    }
}

export const goToAudit = (history: History, redirect?: boolean) => {
    (redirect ? history.replace : history.push)(ROUTE.AUDITOR);
};

export const goToAdmin = (history: History) => {
    history.push(ROUTE_ADMIN.PROMO_CAMPAIGN);
};

export const goToUserManager = (history: History, redirect?: boolean) => {
    (redirect ? history.replace : history.push)(ROUTE_USER_MANAGER.USERS);
};

export const goToPartner = (history: History, redirect?: boolean) => {
    (redirect ? history.replace : history.push)(ROUTE_PARTNER.USERS);
};

export const goToProduct = (history: History) => {
    history.push(ROUTE_OWNER.PROMO_CAMPAIGN);
};

export const goApp = (history: History, role: string) => {
    switch (role) {
        case ROLES.AUDITOR:
            goToAudit(history);
            break;
        case ROLES.ADMIN:
            goToAdmin(history);
            break;
        case ROLES.USER_MANAGER:
            goToUserManager(history);
            break;
        case ROLES.PARTNER:
            goToPartner(history);
            break;
        case ROLES.PRODUCT_OWNER:
        default:
            goToProduct(history);
    }
};

export function getLinkForCreatePromoCampaign() {
    switch (getRole()) {
        case ROLES.ADMIN:
            return `${ ROUTE_ADMIN.PROMO_CAMPAIGN }${ PROMO_CAMPAIGN_PAGES.ADD_PROMO_CAMPAIGN }`;
        case ROLES.PRODUCT_OWNER:
            return `${ ROUTE_OWNER.PROMO_CAMPAIGN }${ PROMO_CAMPAIGN_PAGES.ADD_PROMO_CAMPAIGN }`;
        default:
            return '';
    }
}

export function getPathForCopyPromoCampaign() {
    switch (getRole()) {
        case ROLES.ADMIN:
            return `${ ROUTE_ADMIN.PROMO_CAMPAIGN }${ PROMO_CAMPAIGN_PAGES.PROMO_CAMPAIGN_COPY }`;
        case ROLES.PRODUCT_OWNER:
            return `${ ROUTE_OWNER.PROMO_CAMPAIGN }${ PROMO_CAMPAIGN_PAGES.PROMO_CAMPAIGN_COPY }`;
        default:
            return '';
    }
}

export function getPathForPromoCampaignInfo() {
    switch (getRole()) {
        case ROLES.ADMIN:
            return `${ ROUTE_ADMIN.PROMO_CAMPAIGN }${ PROMO_CAMPAIGN_PAGES.PROMO_CAMPAIGN_INFO }`;
        case ROLES.PRODUCT_OWNER:
            return `${ ROUTE_OWNER.PROMO_CAMPAIGN }${ PROMO_CAMPAIGN_PAGES.PROMO_CAMPAIGN_INFO }`;
        default:
            return '';
    }
}

export function getPathForCreatePromoCampaignVisibititySetting() {
    switch (getRole()) {
        case ROLES.ADMIN:
            return `${ ROUTE_ADMIN.PROMO_CAMPAIGN }${ PROMO_CAMPAIGN_PAGES.VISIBILITY_SETTINGS }/create`;
        case ROLES.PRODUCT_OWNER:
            return `${ ROUTE_OWNER.PROMO_CAMPAIGN }${ PROMO_CAMPAIGN_PAGES.VISIBILITY_SETTINGS }/create`;
        default:
            return '';
    }
}

export function getPathForConsentInfo() {
    switch (getRole()) {
        case ROLES.ADMIN:
            return `${ ROUTE_ADMIN.CONSENTS }${ CONSENTS_PAGES.INFO_CONSENT }`;
        case ROLES.PRODUCT_OWNER:
            return `${ ROUTE_OWNER.CONSENTS }${ CONSENTS_PAGES.INFO_CONSENT }`;
        default:
            return '';
    }
}

export function getPathForConsentsList() {
    switch (getRole()) {
        case ROLES.ADMIN:
            return ROUTE_ADMIN.CONSENTS;
        case ROLES.PRODUCT_OWNER:
            return ROUTE_OWNER.CONSENTS;
        default:
            return '';
    }
}

export function getLinkForPromoCampaignPage() {
    switch (getRole()) {
        case ROLES.ADMIN:
            return ROUTE_ADMIN.PROMO_CAMPAIGN;
        case ROLES.PRODUCT_OWNER:
            return ROUTE_OWNER.PROMO_CAMPAIGN;
        default:
            return '';
    }
}
