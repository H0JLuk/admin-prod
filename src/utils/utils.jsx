
import { ExclamationCircleOutlined } from '@ant-design/icons';
import { Modal } from 'antd';
import React from 'react';

export function highlightSearchingText(value = '', searchValue = '', highlightClassName) {
    if (!searchValue || !value) {
        return <span>{ value }</span>;
    }

    const newValue = [];
    const index = value
        .trim()
        .toLocaleUpperCase()
        .indexOf(
            searchValue
                .trim()
                .toLocaleUpperCase()
            );

    if (index + 1) {
        newValue.push(value.substring(0, index));
        newValue.push(
            <span key={ index } className={ highlightClassName }>
                { value.substring(index, index + searchValue.length) }
            </span>
        );
        newValue.push(value.substring(index + searchValue.length));
    } else {
        newValue.push(value);
    }

    return <span>{ newValue }</span>;
}

export function getStringOptionValue({ parentName = '', name = '' } = {}) {
    if (!parentName) {
        return name ?? '';
    }

    return [parentName, name].join(', ');
}

export function sleep(ms) {
    return new Promise(resolve => setTimeout(resolve, ms));
}

/**
 * @param {import('antd/lib/modal').ModalFuncProps} data
 */
export function confirmModal({
    onOk,
    title = 'Вы уверены?',
    icon: customIcon,
    content = '',
    okText = 'Да',
    cancelText = 'Отмена',
    centered = true,
    ...restOptions
}) {
    const icon = customIcon ? customIcon : <ExclamationCircleOutlined />;
    Modal.confirm({ onOk, title, icon, content, okText, cancelText, centered, ...restOptions });
}

/**
 * @param {import('antd/lib/modal').ModalFuncProps} data
 */
export function successModal({
    onOk,
    title = 'Готово',
    okText = 'ОК',
    cancelText = 'Отмена',
    centered = true,
    cancelButtonProps = { hidden: true },
    maskClosable = false,
    ...restOptions
}) {
    Modal.success({ onOk, title, okText, cancelText, centered, cancelButtonProps, maskClosable, ...restOptions });
}

/**
 * @param {import('antd/lib/modal').ModalFuncProps} data
 */
export function errorModal({
    onOk,
    title = 'Готово',
    okText = 'ОК',
    cancelText = 'Отмена',
    centered = true,
    cancelButtonProps = { hidden: true },
    maskClosable = false,
    ...restOptions
}) {
    Modal.error({ onOk, title, okText, cancelText, centered, cancelButtonProps, maskClosable, ...restOptions });
}