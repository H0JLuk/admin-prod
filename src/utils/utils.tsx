import React from 'react';
import { DEFAULT_SLEEP_TIME } from '@constants/common';
import { Modal, ModalFuncProps } from 'antd';
import { ExclamationCircleOutlined } from '@ant-design/icons';

export function highlightSearchingText(value = '', searchValue = '', highlightClassName = '') {
    if (!searchValue || !value) {
        return <span>{value}</span>;
    }

    const newValue = [];
    const index = value
        .trim()
        .toLocaleUpperCase()
        .indexOf(
            searchValue
                .trim()
                .toLocaleUpperCase(),
        );

    if (index + 1) {
        newValue.push(value.substring(0, index));
        newValue.push(
            <span key={index} className={highlightClassName}>
                {value.substring(index, index + searchValue.length)}
            </span>,
        );
        newValue.push(value.substring(index + searchValue.length));
    } else {
        newValue.push(value);
    }

    return <span>{newValue}</span>;
}

export async function requestsWithMinWait<T>(promise: Promise<T>, minWait = DEFAULT_SLEEP_TIME): Promise<T> {
    const [result] = await Promise.all([promise, sleep(minWait)]);
    return result;
}

export function getStringOptionValue({ parentName = '', name = '' } = {}) {
    if (!parentName) {
        return name ?? '';
    }

    return [parentName, name].join(', ');
}

export function sleep(ms: number): Promise<void> {
    return new Promise(resolve => setTimeout(resolve, ms));
}

export function confirmModal({
    onOk,
    title = 'Вы уверены?',
    icon: customIcon,
    content = '',
    okText = 'Да',
    cancelText = 'Отмена',
    centered = true,
    ...restOptions
}: ModalFuncProps) {
    const icon = customIcon ? customIcon : <ExclamationCircleOutlined />;
    Modal.confirm({ onOk, title, icon, content, okText, cancelText, centered, ...restOptions });
}

export function successModal({
    onOk,
    title = 'Готово',
    okText = 'ОК',
    cancelText = 'Отмена',
    centered = true,
    cancelButtonProps = { hidden: true },
    maskClosable = false,
    ...restOptions
}: ModalFuncProps) {
    Modal.success({ onOk, title, okText, cancelText, centered, cancelButtonProps, maskClosable, ...restOptions });
}

export function errorModal({
    onOk,
    title = 'Готово',
    okText = 'ОК',
    cancelText = 'Отмена',
    centered = true,
    cancelButtonProps = { hidden: true },
    maskClosable = false,
    ...restOptions
}: ModalFuncProps) {
    Modal.error({ onOk, title, okText, cancelText, centered, cancelButtonProps, maskClosable, ...restOptions });
}
