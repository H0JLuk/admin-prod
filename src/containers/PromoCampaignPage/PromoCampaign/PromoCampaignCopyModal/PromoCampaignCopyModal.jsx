import React, { useState } from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
import { useHistory } from 'react-router-dom';
import Modal from 'antd/lib/modal/Modal';
import { Button } from 'antd';
import ExclamationCircleOutlined from '@ant-design/icons/ExclamationCircleOutlined';
import { getPathForCopyPromoCampaign } from '../../../../utils/appNavigation';

const MODAL_CANCEL = 'Нет, настроить заново';
const MODAL_OK = 'Да, скопировать';
const MODAL_CONTENT = 'Хотите скопировать промо-кампанию вместе с настройками видимости?';

const modalPrefixClass = 'ant-modal-confirm';

const PromoCampaignCopyModal = ({
    promoCampaignData,
    children,
}) => {
    const [isModalOpen, setIsModalOpen] = useState(false);
    const history = useHistory();

    if (!promoCampaignData) {
        return children;
    }

    try {
        React.Children.only(children);
    } catch (e) {
        console.error('`children` of `PromoCampaignCopyModal` component  must be single reactNode');
        return null;
    }

    const openModal = () => setIsModalOpen(true);
    const closeModal = () => setIsModalOpen(false);

    const onCopyPromoCampaign = async (copyVisibilitySettings = true) => {
        const routeState = {
            promoCampaign: promoCampaignData,
            copyVisibilitySettings,
        };
        history.push(getPathForCopyPromoCampaign(), routeState);
    };

    const copyWithSettings = () => onCopyPromoCampaign();
    const copyWithoutSettings = () => onCopyPromoCampaign(false);

    return (
        <>
            { React.cloneElement(children, { onClick: openModal }) }
            <Modal
                visible={ isModalOpen }
                width={ 450 }
                onCancel={ closeModal }
                footer={ null }
                className={ cn(modalPrefixClass, `${ modalPrefixClass }-confirm`) }
                destroyOnClose
                centered
            >
                <div className={ `${ modalPrefixClass }-body-wrapper` }>
                    <div className={ `${ modalPrefixClass }-body` }>
                        <ExclamationCircleOutlined />
                        <span className={ `${ modalPrefixClass }-title` }>
                            У промо-кампании
                            <b> { promoCampaignData.name } </b>
                            есть настройки видимости
                        </span>
                        <div className={ `${ modalPrefixClass }-content` }>
                            { MODAL_CONTENT }
                        </div>
                    </div>
                    <div className={ `${ modalPrefixClass }-btns` }>
                        <Button onClick={ copyWithoutSettings }
                        >
                            { MODAL_CANCEL }
                        </Button>
                        <Button
                            type="primary"
                            onClick={ copyWithSettings }
                        >
                            { MODAL_OK }
                        </Button>
                    </div>
                </div>
            </Modal>
        </>
    );
};

PromoCampaignCopyModal.propTypes = {
    promoCampaignData: PropTypes.object.isRequired,
    children: PropTypes.node.isRequired,
};

export default PromoCampaignCopyModal;
