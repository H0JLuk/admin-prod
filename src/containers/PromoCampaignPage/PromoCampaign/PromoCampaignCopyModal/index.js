import PromoCampaignCopyModal from './PromoCampaignCopyModal';

export default PromoCampaignCopyModal;
