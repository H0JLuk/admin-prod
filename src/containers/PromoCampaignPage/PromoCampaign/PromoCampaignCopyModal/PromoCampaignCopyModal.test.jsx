import React from 'react';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import { render, fireEvent } from '@testing-library/react';
import { shallow } from 'enzyme';
import PromoCampaignCopyModal from './PromoCampaignCopyModal';
import { getPathForCopyPromoCampaign } from '../../../../utils/appNavigation';

const promoCampaignTestData = {
    testData: 'testData',
};
const testPathCopyPromoCampaign = '/test/path/to-copy/promoCampaign';
jest.mock('../../../../utils/appNavigation', () => ({
    getPathForCopyPromoCampaign: jest.fn(),
}));


describe('<PromoCampaignCopyModal /> tests.', () => {

    it('should render only children', () => {
        console.error = jest.fn();
        const wrapper = shallow(
            <PromoCampaignCopyModal>
                <div id="test">test</div>
            </PromoCampaignCopyModal>
        );

        expect(wrapper.find('Modal')).toHaveLength(0);
        expect(wrapper.html()).toMatchSnapshot();
    });

    it('should return null', () => {
        const wrapper = shallow(
            <PromoCampaignCopyModal promoCampaignData={ promoCampaignTestData }>
                <div id="test">test</div>
                <div id="test">test</div>
            </PromoCampaignCopyModal>
        );

        expect(wrapper.html()).toBeNull();
    });

    describe('render with modal.', () => {
        const wrapper = shallow(
            <PromoCampaignCopyModal promoCampaignData={ promoCampaignTestData }>
                <div id="test">test</div>
            </PromoCampaignCopyModal>
        );

        it('should render modal and children', () => {
            expect(wrapper.find('Modal')).toHaveLength(1);
            expect(wrapper.html()).toMatchSnapshot();
        });

        it('should open modal and close', () => {
            expect(wrapper.find('Modal').prop('visible')).toBe(false);
            wrapper.find('div#test').simulate('click');
            expect(wrapper.find('Modal').prop('visible')).toBe(true);
            wrapper.find('Modal').prop('onCancel')();
            expect(wrapper.find('Modal').prop('visible')).toBe(false);
        });

        it('should open modal and history.push to copy promoCampaign page with copy visibility settings', () => {
            const history = createMemoryHistory();
            const spyOnHistory = jest.spyOn(history, 'push');
            getPathForCopyPromoCampaign.mockReturnValueOnce('/test/path/to-copy/promoCampaign');

            const { getByTestId } = render(
                <Router history={ history }>
                    <PromoCampaignCopyModal promoCampaignData={ promoCampaignTestData }>
                        <div data-testid="test">test</div>
                    </PromoCampaignCopyModal>
                </Router>
            );

            fireEvent.click(getByTestId('test'));
            expect(document.body.querySelector('.ant-modal-root')).toBeInTheDocument();

            fireEvent.click(document.body.querySelector('button.ant-btn-primary'));
            expect(spyOnHistory).toHaveBeenCalledWith(
                testPathCopyPromoCampaign,
                {
                    promoCampaign: promoCampaignTestData,
                    copyVisibilitySettings: true,
                }
            );
        });

        it('should open modal and history.push to copy promoCampaign page without copy visibility settings', () => {
            const history = createMemoryHistory();
            const spyOnHistory = jest.spyOn(history, 'push');
            getPathForCopyPromoCampaign.mockReturnValueOnce('/test/path/to-copy/promoCampaign');

            const { getByTestId } = render(
                <Router history={ history }>
                    <PromoCampaignCopyModal promoCampaignData={ promoCampaignTestData }>
                        <div data-testid="test">test</div>
                    </PromoCampaignCopyModal>
                </Router>
            );

            fireEvent.click(getByTestId('test'));
            expect(document.body.querySelector('.ant-modal-root')).toBeInTheDocument();

            fireEvent.click(document.body.querySelector('button.ant-btn:not(.ant-btn-primary)'));
            expect(spyOnHistory).toHaveBeenCalledWith(
                testPathCopyPromoCampaign,
                {
                    promoCampaign: promoCampaignTestData,
                    copyVisibilitySettings: false,
                }
            );
        });
    });

});
