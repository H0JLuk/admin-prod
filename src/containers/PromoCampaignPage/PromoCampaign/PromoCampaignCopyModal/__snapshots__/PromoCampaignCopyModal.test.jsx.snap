// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PromoCampaignCopyModal /> tests. render with modal. should render modal and children 1`] = `"<div id=\\"test\\">test</div>"`;

exports[`<PromoCampaignCopyModal /> tests. should render only children 1`] = `"<div id=\\"test\\">test</div>"`;
